crates/wasi/src/preview1.rs

```diff
@@ -2321,6 +2321,29 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiP1Ctx {
             // Indefinite sleeping is not supported in preview1.
             return Err(types::Errno::Inval.into());
         }
+
+        if nsubscriptions == 1 {
```

Comment:
Can you please leave a comment here describing why this special case exists - specifically that it was experimentally determined to result in more consistent sleep times as required for real-time use cases?

Reply:
Hi @pchickey ,

Yes, I added a comment for more explanation. Hopefully that helps us understand this special case.